compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -388,13 +389,11 @@ pub fn from_fn_attrs<'ll, 'tcx>(
             .tcx
             .get_attr(instance.def_id(), sym::target_feature)
             .map_or_else(|| cx.tcx.def_span(instance.def_id()), |a| a.span);
-        let msg = format!(
-            "the target features {} must all be either enabled or disabled together",
-            f.join(", ")
-        );
-        let mut err = cx.tcx.sess.struct_span_err(span, &msg);
-        err.help("add the missing features in a `target_feature` attribute");
-        err.emit();
+        cx.tcx
+            .sess
+            .create_err(TargetFeatureDisableOrEnable { features: f, span: Some(span) })
+            .subdiagnostic(MissingFeatures)
```

Comment:
Why can't you just add a `#[help]` to the type of `TargetFeatureDisableOrEnable`?

Reply:
`TargetFeatureDisableOrEnable` is present in llvm_util.rs, but without the `help`. I was thinking of splitting them into separate diagnostics, and then I could just add `#[help]` as you proposed. 
https://github.com/rust-lang/rust/blob/2227d5117d0d8fe9d897bb3a2d020e75bb3edad5/compiler/rustc_codegen_llvm/src/llvm_util.rs#L497-L502

Reply:
Other solution could be usage of enums instead of struct here. That would boil down to the same problem as the other comment here.